src/components/datepicker/datePicker.js

```diff
@@ -260,22 +321,62 @@
    * @param {!angular.NgModelController} ngModelCtrl
    */
   DatePickerCtrl.prototype.configureNgModel = function(ngModelCtrl) {
-    this.ngModelCtrl = ngModelCtrl;
-
     var self = this;
-    ngModelCtrl.$render = function() {
-      var value = self.ngModelCtrl.$viewValue;
 
-      if (value && !(value instanceof Date)) {
-        throw Error('The ng-model for md-datepicker must be a Date instance. ' +
+    if (ngModelCtrl == null && self.begin != undefined && self.end != undefined) {
```

Comment:
This function should be broken up into smaller functions; at least one for the ngModel setup and one for the range setup, with possibly other functions for anything shared between them.